src/Core/Scene/Entity.hh

```diff
@@ -0,0 +1,44 @@
+#ifndef SCENE_ENTITY_HH
+#define SCENE_ENTITY_HH
+
+#include "Scene.hh"
+
+#include "esppch.hh"
+
+namespace esp
+{
+  class Entity
+  {
+   private:
+    entt::entity m_handle{ 0 };
+    std::shared_ptr<Scene> m_scene;
+
+   public:
+    template<typename T, typename... Args> T& add_component(Args&&... args)
+    {
+      ESP_ASSERT(!has_component<T>(), "Entity already has component")
```

Comment:
Mam uwagę, która odnosi się całościowo do silnika, ale tu jest dobry przykład. Naszpikowaliśmy kod assertami, które przy wykryciu błędu wywracają cały program bez możliwości obsłużenia tego. Jest to wygodne podczas developmentu bo odrazu widać, że coś nie działa, ale z perspektywy użytkownika jest to problematyczne.

Reply:
Noo to prawda. A jakbyś to inaczej zrobił?

Reply:
Proponuję albo rzucać jakiś wyjątek, albo moglibyśmy stworzyć generyczną klasę ```Result```, która zawiera w sobie kod wyniku i opcjonalnie zwracaną wartość. Mielibyśmy jakiś enum z kodami, np success, not_found itp

Reply:
No spoko pomysł zachęcam to zrobienia czegoś takiego w przyszłości:D